flyteadmin/pkg/async/shared.go

```diff
@@ -10,6 +11,24 @@ import (
 // RetryDelay indicates how long to wait between restarting a subscriber connection in the case of network failures.
 var RetryDelay = 30 * time.Second
 
+func RetryOnSpecificErrorCodes(totalAttempts int, delay time.Duration, f func() (*http.Response, error), IsErrorCodeRetryable func(*http.Response) bool) error {
```

Comment:
since this is specifically for handling http requests given the function signature, what if we renamed it to something like 

RetryHTTPRequest...

Reply:
Yes definitely!
[Context on previous comments](https://github.com/flyteorg/flyte/pull/4259#discussion_r1364617660) on the last PR (i borked it)

Reply:
Trying to migrate to the retry.OnError now, @katrogan @pmahindrakar-oss I'll let you know when i push those changes

Reply:
not at all! just a comment on naming for if we choose to use this custom method

Reply:
if this is unused, mind removing too?